class.exportcontroller.php

```diff
@@ -127,7 +127,7 @@ public function handleInfoForm() {
         $this->dbInfo = array(
             'dbhost' => $_POST['dbhost'],
             'dbuser' => $_POST['dbuser'],
-            'dbpass' => $_POST['dbpass'],
```

Comment:
Are you sure this doesn't break anything? Usually we only escape at time of output; this looks like storage that would result in an incorrect password.

Reply:
I am using `htmlspecialchars_decode` because we might receive an escaped value because of this change.

https://github.com/vanilla/porter/pull/185/files#diff-73042ed68bd24658b1a6e1e21e3d4f6fR148

Reply:
I might be wrong, I will investigate a bit further.

Reply:
I concur with @linc on this one. When you `htmlspecialchars()` inside `input value="here"` it doesn't get posted back escaped. It's just for outputting.

Reply:
Annnnd I was wrong, I misjudged how these encoded characters were treated by the browser. I will revert this change.

However, I have the following question. Is it necessary to output the value in the form? I feel like we could just put an empty value, but maybe there's something I don't know.

```html
<input class="InputBox" type="password" name="dbpass" value="<?php echo htmlspecialchars(getValue('dbpass')) ?>"/>
```

Reply:
Outputting the value is just a convenience thing so you don't have to retype all the time. Many systems do this, but omit the password which I think is a sensible tradeoff.